tests/test_wagon.py

```diff
@@ -0,0 +1,686 @@
+        else:
+            opts_and_args.append(opt)
+    return clicktest.CliRunner().invoke(getattr(wagon, func), opts_and_args)
+
+
+class TestBase(testtools.TestCase):
+
+    def test_run(self):
+        proc = wagon.run('uname')
+        self.assertEqual(0, proc.returncode)
+
+    def test_run_bad_command(self):
+        proc = wagon.run('suname')
+        self.assertEqual(1 if wagon.IS_WIN else 127, proc.returncode)
+
+    def test_download_file(self):
+        wagon._download_file(TEST_TAR, 'file')
+        try:
+            os.remove('file')
+        except:
+            self.fail()
+
+    def test_download_file_missing(self):
+        e = self.assertRaises(
+            IOError,
+            wagon._download_file,
+            'http://www.google.com/x.tar.gz',
+            'file')
+        self.assertIn("'http error', 404, 'Not Found'", str(e))
+
+    def test_download_bad_url(self):
+        e = self.assertRaises(
+            IOError, wagon._download_file, 'something', 'file')
+        if wagon.IS_WIN:
+            self.assertIn(
+                "The system cannot find the file specified: 'something'",
+                str(e))
+        else:
+            self.assertIn("No such file or directory: 'something'", str(e))
+
+    def test_download_missing_path(self):
+        e = self.assertRaises(
+            IOError, wagon._download_file, TEST_TAR, 'x/file')
+        self.assertIn('No such file or directory', e)
+
+    def test_download_no_permissions(self):
+        if wagon.IS_WIN:
+            self.skipTest('Irrelevant on Windows.')
+        e = self.assertRaises(IOError, wagon._download_file, TEST_TAR, '/file')
+        self.assertIn('Permission denied', str(e))
+
+    def test_tar(self):
+        os.makedirs('dir')
+        with open('dir/content.file', 'w') as f:
+            f.write('CONTENT')
+        wagon._tar('dir', 'tar.file')
+        shutil.rmtree('dir')
+        self.assertTrue(tarfile.is_tarfile('tar.file'))
+        with closing(tarfile.open('tar.file', 'r:gz')) as tar:
+            members = tar.getnames()
+            self.assertIn('dir/content.file', members)
+        os.remove('tar.file')
+
+    def test_tar_no_permissions(self):
+        if wagon.IS_WIN:
+            self.skipTest("Irrelevant on Windows.")
+        tmpdir = tempfile.mkdtemp()
+        try:
+            e = self.assertRaises(IOError, wagon._tar, tmpdir, '/file')
+            self.assertIn("Permission denied: '/file'", str(e))
+        finally:
+            shutil.rmtree(tmpdir)
+
+    def test_tar_missing_source(self):
+        e = self.assertRaises(OSError, wagon._tar, 'missing', 'file')
+        if wagon.IS_WIN:
+            self.assertIn(
+                "The system cannot find the file specified: 'missing'",
+                str(e))
+        else:
+            self.assertIn("No such file or directory: 'missing'", str(e))
+        os.remove('file')
+
+    def test_make_virtualenv_illegal_dir(self):
+        if wagon.IS_WIN:
```

Comment:
to make the test code cleaner take advantage of testtools.skipIf decorator. like so:
@testtools.skipIf(wagon.IS_WIN, 'Irrelevant on Windows.')